keeper-gradle-plugin/src/main/java/com/slack/keeper/InferAndroidTestKeepRules.kt

```diff
@@ -101,29 +101,35 @@ public abstract class InferAndroidTestKeepRules : JavaExec() {
 
     enableAssertions = enableAssertionsProperty.get()
     args = when (traceReferencesEnabled.get()) {
-      false -> listOf(
-          "--keeprules",
-          androidJar.get().asFile.absolutePath,
-          appTargetJar.get().asFile.absolutePath,
-          androidTestSourceJar.get().asFile.absolutePath
-      ).also {
-        // print-uses is using its output to print rules
-        standardOutput = outputProguardRules.asFile.get().outputStream().buffered()
-      }
-      true -> listOf<Pair<String, String?>>(
-          "--keep-rules" to "",
-          "--lib" to androidJar.get().asFile.absolutePath,
-          "--lib" to androidTestJar.orNull?.asFile?.absolutePath,
-          "--target" to appTargetJar.get().asFile.absolutePath,
-          "--source" to androidTestSourceJar.get().asFile.absolutePath,
-          "--output" to outputProguardRules.get().asFile.absolutePath
-      ).map { if (it.second != null) listOf(it.first, it.second) else listOf() }
-          .reduce { acc, any -> acc + any } + traceReferencesArgs.getOrElse(listOf())
+      false -> genPrintUsesArgs()
+      true -> genTraceReferencesArgs()
     }
 
     super.exec()
   }
 
+  private fun genPrintUsesArgs(): List<String> =
+    listOf(
+      "--keeprules",
+      androidJar.get().asFile.absolutePath,
+      appTargetJar.get().asFile.absolutePath,
+      androidTestSourceJar.get().asFile.absolutePath
+    ).also {
+      // print-uses is using its output to print rules
+      standardOutput = outputProguardRules.asFile.get().outputStream().buffered()
+    }
+
+  private fun genTraceReferencesArgs(): List<String?> =
+    listOf<Pair<String, String?>>(
+      "--keep-rules" to "",
+      "--lib" to androidJar.get().asFile.absolutePath,
+      "--lib" to androidTestJar.orNull?.asFile?.absolutePath,
+      "--target" to appTargetJar.get().asFile.absolutePath,
+      "--source" to androidTestSourceJar.get().asFile.absolutePath,
+      "--output" to outputProguardRules.get().asFile.absolutePath
+    ).map { if (it.second != null) listOf(it.first, it.second) else listOf() }
+      .reduce { acc, any -> acc + any } + traceReferencesArgs.getOrElse(listOf())
```

Comment:
This logic is still confusing, let's comment or extract this too. The rest of it was actually fine, this was the part I wanted to clean up

Reply:
I agree with you, but I could not find a way to put this + in a new line until afd10b2 
I also added a comment because the map/reduce is probably not the easiest construction to read if followed by another instruction.